{"Modified":"2022-01-05T16:45:00","Published":"2021-12-26T22:15:00","access":{"authentication":"NONE","complexity":"LOW","vector":"NETWORK"},"assigner":"cve@mitre.org","capec":[],"cvss":5,"cvss-time":"2022-01-05T16:45:00","cvss-vector":"AV:N/AC:L/Au:N/C:N/I:N/A:P","cwe":"CWE-416","id":"CVE-2021-45718","impact":{"availability":"PARTIAL","confidentiality":"NONE","integrity":"NONE"},"last-modified":"2022-01-05T16:45:00","references":["https://rustsec.org/advisories/RUSTSEC-2021-0128.html","https://raw.githubusercontent.com/rustsec/advisory-db/main/crates/rusqlite/RUSTSEC-2021-0128.md"],"summary":"An issue was discovered in the rusqlite crate 0.25.x before 0.25.4 and 0.26.x before 0.26.2 for Rust. rollback_hook has a use-after-free.","vulnerable_configuration":[{"id":"cpe:2.3:a:rusqlite_project:rusqlite:0.25.0:*:*:*:*:rust:*:*","title":"cpe:2.3:a:rusqlite_project:rusqlite:0.25.0:*:*:*:*:rust:*:*"},{"id":"cpe:2.3:a:rusqlite_project:rusqlite:0.25.1:*:*:*:*:rust:*:*","title":"cpe:2.3:a:rusqlite_project:rusqlite:0.25.1:*:*:*:*:rust:*:*"},{"id":"cpe:2.3:a:rusqlite_project:rusqlite:0.25.2:*:*:*:*:rust:*:*","title":"cpe:2.3:a:rusqlite_project:rusqlite:0.25.2:*:*:*:*:rust:*:*"},{"id":"cpe:2.3:a:rusqlite_project:rusqlite:0.25.3:*:*:*:*:rust:*:*","title":"cpe:2.3:a:rusqlite_project:rusqlite:0.25.3:*:*:*:*:rust:*:*"},{"id":"cpe:2.3:a:rusqlite_project:rusqlite:0.26.0:*:*:*:*:rust:*:*","title":"cpe:2.3:a:rusqlite_project:rusqlite:0.26.0:*:*:*:*:rust:*:*"},{"id":"cpe:2.3:a:rusqlite_project:rusqlite:0.26.1:*:*:*:*:rust:*:*","title":"cpe:2.3:a:rusqlite_project:rusqlite:0.26.1:*:*:*:*:rust:*:*"}],"vulnerable_configuration_cpe_2_2":[],"vulnerable_product":["cpe:2.3:a:rusqlite_project:rusqlite:0.25.0:*:*:*:*:rust:*:*","cpe:2.3:a:rusqlite_project:rusqlite:0.25.1:*:*:*:*:rust:*:*","cpe:2.3:a:rusqlite_project:rusqlite:0.25.2:*:*:*:*:rust:*:*","cpe:2.3:a:rusqlite_project:rusqlite:0.25.3:*:*:*:*:rust:*:*","cpe:2.3:a:rusqlite_project:rusqlite:0.26.0:*:*:*:*:rust:*:*","cpe:2.3:a:rusqlite_project:rusqlite:0.26.1:*:*:*:*:rust:*:*"]}